jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/WebAppContext.java

```diff
@@ -1449,13 +1450,11 @@ public URL getResource(String path) throws MalformedURLException
                 return null;
 
             // Should we go to the original war?
-            if (resource.isDirectory() && resource instanceof ResourceCollection && !WebAppContext.this.isExtractWAR())
+            if (resource.isDirectory() && !WebAppContext.this.isExtractWAR())
             {
-                List<Resource> resources = ((ResourceCollection)resource).getResources();
-                for (int i = resources.size(); i-- > 0; )
+                for (Resource r: resource)
                 {
-                    Resource r = resources.get(i);
-                    if (r.getName().startsWith("jar:file"))
+                    if (FileID.isWebArchive(r.getFileName()))
```

Comment:
Seems like a different logic, as before the _last_ hit in any archive would be returned, now it returns the _first_ hit in the war?

Reply:
It is, but this logic can be simplified even further.
Stay tuned for update.